src/transformers/generation/streamers.py

```diff
@@ -105,13 +106,13 @@ def put(self, value):
             self.token_cache = []
             self.print_len = 0
         # If the last token is a CJK character, we print the characters.
-        elif len(text) > 0 and self._is_chinese_char(ord(text[-1])):
+        elif len(text) > 0 and (text[-1] in string.printable or self._is_chinese_char(ord(text[-1]))):
```

Comment:
This should explicitly avoid punctuation, as it may be some form of formatting. E.g.: what if `text[-1]` is `\` (part of `string.printable`) and the model is trying to generate `\n` (linebreak)?

In other words, print if `text[-1]` is a digit, a character, or whitespace 🤗